api/src/main/java/org/apache/iceberg/expressions/ExpressionUtil.java

```diff
@@ -600,6 +612,12 @@ private static String sanitizeString(CharSequence value, long now, int today) {
       } else if (TIMESTAMPTZ.matcher(value).matches()) {
         Literal<Long> ts = Literal.of(value).to(Types.TimestampType.withZone());
         return sanitizeTimestamp(ts.value(), now);
+      } else if (TIMESTAMPNS.matcher(value).matches()) {
+        Literal<Long> ts = Literal.of(value).to(Types.TimestampNanoType.withoutZone());
+        return sanitizeTimestamp(ts.value() / 1000, now);
+      } else if (TIMESTAMPTZNS.matcher(value).matches()) {
+        Literal<Long> ts = Literal.of(value).to(Types.TimestampNanoType.withZone());
+        return sanitizeTimestamp(ts.value() / 1000, now);
```

Comment:
Is it safe to divide or do we need to use `Math.floorDiv`? I think we use that function in other places.

Reply:
Using `Math.floorDiv` now.